pkg/build/api/validation/validation.go

```diff
@@ -503,6 +503,7 @@ func validateSourceStrategy(strategy *buildapi.SourceBuildStrategy, fldPath *fie
 	allErrs = append(allErrs, validateFromImageReference(&strategy.From, fldPath.Child("from"))...)
 	allErrs = append(allErrs, validateSecretRef(strategy.PullSecret, fldPath.Child("pullSecret"))...)
 	allErrs = append(allErrs, ValidateStrategyEnv(strategy.Env, fldPath.Child("env"))...)
+	allErrs = append(allErrs, validateRuntimeImage(strategy, fldPath.Child("runtimeImage"))...)
```

Comment:
@bparees the validation that i've added here applies to the for the build object, so you cannot do `oc create -f build.json` with both incremental=true and a runtime image defined. 
a bc that can test this is here https://gist.githubusercontent.com/PI-Victor/16f030f52ef6127149a8f1a12d7cc0b9/raw/a62bda283f6e6d0f84d7eb6ac8eee2fc968d2b31/gistfile1.txt

both validateBuild and validateBuildConfig run the validateCommonSpec which runs the below code, so it's covered.